include/cctz/civil_time_detail.h

```diff
@@ -128,15 +130,26 @@ CONSTEXPR_F fields n_day(year_t y, month_t m, diff_t d, diff_t cd,
     }
   }
   if (d > 365) {
+    // Index into Gregorian 400 year cycle.
+    int yi = year_index(ey, m);
     for (;;) {
-      int n = days_per_century(ey, m);
+      int n = days_per_century(yi);
       if (d <= n) break;
+      yi += 100;
+      // Maintain yi being ey mod 400.
+      if (yi >= 400) {
```

Comment:
Done, though I do find this form slightly less clear.

Reply:
I was happy with the old form of these ...
```
  yi += 100;
  if (yi >= 400) yi -= 400;
```
where there is per-loop state, so please do change them back.  That is, it was only in the non-looping, functional `year_index()` where I think it is nicer to avoid state.

Reply:
Indeed, let's make the `year_index()` local variable `const`, like the ones in `days_per_century()` and `days_per_4years()` used to be.  Thanks.

Reply:
made yi in year_index() const

Reply:
split increment and fixup into two lines.